ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -274,6 +270,23 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
     isStackedBar,
   ]);
 
+  // Update adjacent charts so tooltip is unpinned when current chart is clicked.
```

Comment:
Are there any low-ish effort unit tests that could be written around some of the tooltip pinning behavior? The logic is getting relatively complex, and I worry it'll be easy to accidentally break things in the future without tests, but I also know testing tooltip-related code can be painful and time consuming.

Reply:
I can definitely write some unit tests, but I'm not sure how useful they'll be since I'll have to mock [echartsInstance.convertFromPixel](https://echarts.apache.org/en/api.html#echartsInstance.convertFromPixel), the integration tests with Playwright are what I really want but the last time I tried, they weren't easy to get working reliably

Reply:
I think it's fine to leave as is for now, so you can get this in. We can take a pass at trying to figure out a non-painful option for testing in a future follow-up.